satellite/contact/endpoint.go

```diff
@@ -43,13 +43,20 @@ func (endpoint *Endpoint) Checkin(ctx context.Context, req *pb.CheckinRequest) (
 	if err != nil {
 		return nil, Error.Wrap(err)
 	}
-	pingNodeSuccess, pingErrorMessage, err := endpoint.pingBack(ctx, req, peerID)
+	nodeID := peerID.ID
+
+	pingNodeSuccess, pingErrorMessage, err := endpoint.pingBack(ctx, req, nodeID)
+	if err != nil {
+		return nil, Error.Wrap(err)
+	}
+
+	err = endpoint.service.peerids.Set(ctx, nodeID, peerID)
```

Comment:
Do we care if the ping back failed or not when setting the peer identity?

Reply:
Probably not - maybe we should set the peer identity first?

Reply:
Updated